core/src/main/scala/org/apache/spark/status/api/v1/AllStagesResource.scala

```diff
@@ -59,6 +59,16 @@ private[v1] object AllStagesResource {
       stageUiData: StageUIData,
       includeDetails: Boolean): StageData = {
 
+    val taskLaunchTimes = stageUiData.taskData.values.map(_.taskInfo.launchTime).filter(_ > 0)
+
+    val firstTaskLaunchedTime: Option[Date] =
+      if (taskLaunchTimes.nonEmpty) {
+        val minTaskLauchTime = taskLaunchTimes.min
+        Some(new Date(minTaskLauchTime))
```

Comment:
super minor, but you don't really need `minTaskLaunchTime`, could just combine it into one line: `Some(new Date(taskLaunchTimes.min))`